Tests/MoyaTests/MultiTargetSpec.swift

```diff
@@ -32,28 +32,43 @@ final class MultiTargetSpec: QuickSpec {
             }
 
             it("uses correct parameters") {
-                if case let .requestParameters(parameters: parameters, encoding: _) = target.task {
-                    expect(parameters["key"] as? String) == "value"
-                    expect(parameters.count) == 1
-                } else {
-                    fail("expected task type `.requestParameters`, was \(String(describing: target.task))")
+                let task = target.task
+                guard case Task.request = task else {
+                    fail("expected task type `.request`, was \(String(describing: target.task))")
+                    return
                 }
+                let allParameters = try? task.allParameters()
+                expect(allParameters).toNot(beNil())
+                let encodable = allParameters!.first?.0
+                expect(encodable).toNot(beNil())
+                let dict = encodable! as? [String: String]
+                expect(dict).toNot(beNil())
+                expect(dict!["key"]) == "value"
+                expect(dict!.count) == 1
             }
 
             it("uses correct parameter encoding.") {
-                if case let .requestParameters(parameters: _, encoding: parameterEncoding) = target.task {
-                    expect(parameterEncoding is JSONEncoding) == true
-                } else {
-                    fail("expected task type `.requestParameters`, was \(String(describing: target.task))")
+                let task = target.task
+                guard case Task.request = task else {
+                    fail("expected task type `.request`, was \(String(describing: target.task))")
+                    return
                 }
+
+                let allParameters = try? task.allParameters()
+                expect(allParameters).toNot(beNil())
+                expect(allParameters!.first?.1).to(beAKindOf(JSONParameterEncoder.self))
             }
 
             it("uses correct method") {
                 expect(target.method) == Method.get
             }
 
             it("uses correct task") {
-                expect(String(describing: target.task)).to(beginWith("requestParameters")) // Hack to avoid implementing Equatable for Task
+                guard case Task.request = target.task else {
+                    fail("expected task type `.request`, was \(String(describing: target.task))")
+                    return
+                }
+                expect(true) == true
```

Comment:
Is this necessary?

Reply:
I'm not familiar with tests yet so I wanted to make sure the test passed. Will it pass by default if no `expect` statement is used?

Reply:
Yeah ... and since the `fail` is the actual `assertion` that is important on this test(we are failing if the task is what we don't expect) ... we could remove that I think